File: ruby/hamming/hamming.rb
class Hamming

  def self.compute(strand1, strand2)
    counter = 0                                   # => 0
    strand1.chars.each_with_index do |letter, i|  # => ["A", "S", "D", "F", "G", "H"]
        if strand1.length != strand2.length
          raise (ArgumentError)
        end
        if strand1.chars[i] != strand2.chars[i]   # => false, false, false, false, false, false
          counter += 1
        end                                       # => nil, nil, nil, nil, nil, nil
      end                                         # => ["A", "S", "D", "F", "G", "H"]
    counter                                       # => 0
  end                                             # => :compute
end                                               # => :compute

File: javascript/hamming/hamming.js
var Hamming = function () {};

Hamming.prototype.compute = function (string1, string2) {
  var counter = 0;
  for(var i = 0; i < string1.length; i ++ ) {
    if (string1[i] != string2[i]) {
      counter += 1
    }
  }
  return counter
}
// Hamming.prototype.compute = function(str1, str2){
//   if(str1.length != str2.length) {throw "DNA strands must be of equal length."}
//
//   var distance = 0;
//
//   for(var i = 0; i < str1.length; i ++){
//     if(str1[i] != str2[i]){
//       distance += 1
//     }
//   }
//   return distance;
// }

module.exports = Hamming;
